src/nvim/eval.c

```diff
@@ -8145,11 +8145,17 @@ static void f_eval(typval_T *argvars, typval_T *rettv)
   if (s != NULL)
     s = skipwhite(s);
 
-  if (s == NULL || eval1(&s, rettv, TRUE) == FAIL) {
+  if (s == NULL || eval1(&s, rettv, true) == FAIL) {
+    char_u *p = s;
```

Comment:
I think you overshot the goal slightly :)

`p` must be left outside the block, it should keep pointing to the input string even after `eval1`ing `s`. As it is the patch is no longer correct.

Also, you cannot simply replace `TRUE` with `true` and `FALSE` with `false`, these variables have type `int` (and sometimes hold values other than `0` and `1`). Unless the conversion to `bool` is trivial it is perhaps better to leave it for another pull request.